devito/finite_differences/coefficients.py

```diff
@@ -176,6 +185,7 @@ def generate_subs(i):
             deriv_order = i.deriv_order
             function = i.function
             dim = i.dimension
+            offset_dim = i.offset_dimension
```

Comment:
Why are `dim` and `offset_dim` both required?

Reply:
So `offset_dim` is the `Add` object? If so, should probably have it's name changed to something related to `index`. And is this `None` when it's not required?

Reply:
It's equivalent to `.dimension` on non staggered grids, but on staggered grids it also includes the half-increment offset introduced by the staggering. `.dimension` would always be `x` regardless as to whether the function is staggered in the x direction or not, but `.offset_dimension` could be `x` or `x+h_x` depending on the stagger

Reply:
The point is that you should never have repeated information and hence the routine should be updated such that this is the case.

Reply:
Wouldn't that require extra conditionals though? Since sometimes `dimension` is required, but other times both `dimension` and `dimension+spacing/2` are required, in which case the latter simply matches the former. This avoids the need to check the value of `offset_dim`, and act accordingly if it is None. Additionally, intuitively, `dimension+offset` where `offset` is 0 is just `dimension`, not `None`